src/app-layer-htp-range.c

```diff
@@ -543,14 +543,24 @@ File *HttpRangeClose(const StreamingBufferConfig *sbcfg, HttpRangeContainerBlock
             // a new range just begins where we ended, append it
             if (range->gap > 0) {
                 // if the range had a gap, begin by it
-                if (FileAppendData(c->container->files, sbcfg, NULL, range->gap) != 0) {
+                uint32_t gap = (uint32_t)range->gap;
+                if (range->gap > UINT32_MAX) {
```

Comment:
If we are downcasting `range->gap` to `uint32_t` first, under what circumstances should it be greater than `UINT32_MAX`?

Reply:
Indeed,
looks like `range->gap > UINT32_MAX` if we had multiple gaps which combine over 4G

Let's redefine `range->gap` to be u32 in the first place

Thanks